src/Build/BackEnd/Components/Logging/BuildEventArgTransportSink.cs

```diff
@@ -145,6 +153,11 @@ public void Consume(BuildEventArgs buildEvent, int sinkId)
                 HaveLoggedBuildFinishedEvent = true;
                 return;
             }
+            else if (buildEvent is BuildCanceledEventArgs)
+            {
+                HaveLoggedBuildFinishedEvent = true;
```

Comment:
```suggestion
                HaveLoggedBuildCanceledEvent = true;
```

. . . but I think the strictness around Started/Finished are special and don't apply here; there are some deeply unfortunate parts of the engine that care about logging events (instead of signaling stuff a more appropriate way), but nothing should care about cancelation that way.

Reply:
I removed this logic from all types.